packages/jupyter/src/utils.ts

```diff
@@ -0,0 +1,212 @@
+import type {
+  Thebe,
+  ThebeServerOptions,
+  ThebeLocalOptions,
+  ThebeBinderOptions,
+} from 'myst-frontmatter';
+import type { CoreOptions, RepoProvider } from 'thebe-core';
+
+export type ExtendedCoreOptions = CoreOptions & {
+  useBinder?: boolean;
+  useJupyterLite?: boolean;
+};
+
+function extractGithubRepoInfo(url: string): { owner: string; repo: string } | null {
+  const pattern = /https?:\/\/github\.com\/([^/]+)\/([^/]+)/;
+  const match = url.match(pattern);
+
+  if (match) {
+    return {
+      owner: match[1],
+      repo: match[2],
+    };
+  }
+
+  return null;
+}
+
+function extractBinderRepoInfo(
+  url: string,
+): { binderUrl: string; repoProvider: string; owner: string; repo: string; ref: string } | null {
+  const pattern = /(https?:\/\/[^/]+(?:\/[^/]+)*?)\/(v\d+)\/([^/]+)\/([^/]+)\/([^/]+)\/([^/]+)/;
+  const match = url.match(pattern);
+
+  if (match) {
+    const repoProviderAbbreviation = match[3];
+    let repoProvider;
+
+    switch (repoProviderAbbreviation) {
+      case 'gh':
+        repoProvider = 'github';
+        break;
+      case 'gl':
+        repoProvider = 'gitlab';
+        break;
+      default:
+        repoProvider = repoProviderAbbreviation;
+    }
+
+    return {
+      binderUrl: match[1],
+      repoProvider: repoProvider,
+      owner: match[4],
+      repo: match[5],
+      ref: match[6],
+    };
+  }
+
+  return null;
+}
+
+function isObject(maybeObject: any) {
+  return typeof maybeObject === 'object' && maybeObject !== null;
+}
+
+export function thebeFrontmatterToOptions(
+  fm: boolean | Thebe | undefined,
+  githubBadgeUrl: string | undefined,
+  binderBadgeUrl: string | undefined,
+): ExtendedCoreOptions | undefined {
+  if (fm === undefined || fm === false) return undefined;
+
+  const {
+    binder,
+    server,
+    lite,
+    local,
+    kernelName,
+    disableSessionSaving,
+    mathjaxConfig,
+    mathjaxUrl,
+  } = (fm as Thebe | undefined) ?? {};
+
+  const thebeOptions: ExtendedCoreOptions = { mathjaxConfig, mathjaxUrl };
+
+  if (disableSessionSaving) {
+    thebeOptions.savedSessionOptions = { enabled: false };
+  }
+
+  // handle thebe.local.*
+  // as local OVERIDES other binder and server settings, handle these first
+  // TODO need to expose NODE_ENV somehow via a loader
+  let NODE_ENV = 'development';
+  if (typeof window !== 'undefined') {
+    NODE_ENV = (window as any).NODE_ENV;
+  }
+  if (NODE_ENV !== 'production' && local) {
```

Comment:
TODO probably for first pass, you need to remove your local key before deploying